### DISCUSSION TITLE
Minimal example of dagmc unstructured mesh source in OpenMC

### DISCUSSION THREAD

**Discussion by shimwell** [svalinn/DAGMC#936] [Q&A] [answered]
Hi all I'm wondering if anyone has a minimal example of  a tet mesh / unstructured mesh / dagmc volume mesh being used as a neutron or photon source term.

Many thanks

**Comment by gonuke** [comment]
I am not sure it's a minimal example, but ParaStell has support for this.  Maybe @connoramoreno can point you to something?

**Comment by connoramoreno** [comment]
What kind of example are you looking for @shimwell ? I can share an OpenMC script that uses a tetrahedral source mesh

**Reply by shimwell** [reply]
That would be super useful @connoramoreno 

Anything would be great. I'm keen to see an example of a tet source and longer term I hope to make an example shut down dose rate simulation that uses tet meshes.

**Reply by connoramoreno** [reply]
Any particular method of sharing that works best for you? I can email, Slack, or I can share the relevant cope snippet here.

**Reply by shimwell** [reply]
Code snippet here would be best for me if you don't mind

**Reply by connoramoreno** [reply, marked as answer]
Sure thing! Here's a typical fusion source definition I use. `strengths` is a list of source strength floats for each tetrahedron, ordered by tetrahedral ID. Let me know if you have any questions!
```
# Define run settings
settings = openmc.Settings()
settings.run_mode = 'fixed source'
settings.photon_transport = True
settings.particles = 10000
settings.batches = 30
settings.survival_biasing = True

# Define source mesh
mesh = openmc.UnstructuredMesh("SourceMesh.h5m", 'moab')

# Define fixed source
src = openmc.IndependentSource()
src.space = openmc.stats.MeshSpatial(
    mesh,
    strengths = strengths,
    volume_normalized = False
)
src.angle = openmc.stats.Isotropic()
src.energy = openmc.stats.Discrete([14.1e6], [1.0])
settings.source = [src]

# Export settings XML
settings.export_to_xml()
```

**Reply by shimwell** [reply]
Many thanks the openmc.stats.MeshSpatial is just what I was looking for. I see it accepts strengths already. Perhaps with a bit of tinkering I can get it to accept energy distributions.

**Reply by gonuke** [reply]
I don't think energy distributions work in the current code but @pshriwise was working on something

**Reply by pshriwise** [reply]
This should be possible now that https://github.com/openmc-dev/openmc/pull/2759 is merged.
